tests/namespaces.bats

```diff
@@ -63,58 +63,37 @@ load helpers
   # Check that with settings that don't require a user namespace, we can request to use a per-container network namespace.
   run_buildah run $RUNOPTS --net=container "$ctr" readlink /proc/self/ns/net
   if [[ $output == $mynetns ]]; then
-      expect_output "[output should not be '$mynetns']"
+      die "[/proc/self/ns/net (--net=container) should not be '$mynetns']"
   fi
 
   run_buildah run $RUNOPTS --net=private "$ctr" readlink /proc/self/ns/net
   if [[ $output == $mynetns ]]; then
-      expect_output "[output should not be '$mynetns']"
+      die "[/proc/self/ns/net (--net=private) should not be '$mynetns']"
   fi
 }
 
-idmapping_check_namespace() {
-  local _uidmapargs=$1
-  local _gidmapargs=$2
-  local _mynamespace=$3
-  local _output=$4
+# Helper for idmapping test: check UID or GID mapping
+# NOTE SIDE EFFECT: sets $rootxid for possible use by caller
+idmapping_check_map() {
+  local _output_idmap=$1
+  local _expect_idmap=$2
+  local _testname=$3
 
-  [ "$_output" != "" ]
-  if [ -z "${_uidmapargs}${_gidmapargs}" ]; then
-    if test "$BUILDAH_ISOLATION" != "chroot" -a "$BUILDAH_ISOLATION" != "rootless" ; then
-      expect_output --from="$_output" "$_mynamespace"
-    fi
-  else
-    [ "$_output" != "$_mynamespace" ]
-  fi
-}
+  [ -n "$_output_idmap" ]
+  local _idmap=$(sed -E -e 's, +, ,g' -e 's,^ +,,g' <<< "${_output_idmap}")
+  expect_output --from="$_idmap" "${_expect_idmap}" "$_testname"
 
-idmapping_check_map() {
-  local _output_uidmap=$1
-  local _output_gidmap=$2
-  local _expect_uidmap=$3
-  local _expect_gidmap=$4
-
-  [ -n "$_output_uidmap" ]
-  local uidmap=$(sed -E -e 's, +, ,g' -e 's,^ +,,g' <<< "${_output_uidmap}")
-  [ -n "$_output_gidmap" ]
-  local gidmap=$(sed -E -e 's, +, ,g' -e 's,^ +,,g' <<< "${_output_gidmap}")
-  echo expected UID map "${_expect_uidmap}", got UID map "${uidmap}", expected GID map "${_expect_gid_map}", got GID map "${gidmap}".
-  expect_output --from="$uidmap" "${_expect_uidmap}"
-  expect_output --from="$gidmap" "${_expect_gidmap}"
-  # these vars are global
-  rootuid=$(sed -E -e 's,^([^ ]*) (.*) ([^ ]*),\2,' <<< "$uidmap")
-  rootgid=$(sed -E -e 's,^([^ ]*) (.*) ([^ ]*),\2,' <<< "$gidmap")
+  # SIDE EFFECT: Global: our caller may want this
+  rootxid=$(sed -E -e 's,^([^ ]*) (.*) ([^ ]*),\2,' <<< "$_idmap")
 }
 
+# Helper for idmapping test: check file permissions
 idmapping_check_permission() {
   local _output_file_stat=$1
   local _output_dir_stat=$2
-  local _output_otherfile_stat=$3
-  local _expect_otherfile_stat=$4
 
   expect_output --from="${_output_file_stat}" "1:1" "Check if a copied file gets the right permissions"
   expect_output --from="${_output_dir_stat}" "0:0" "Check if a copied directory gets the right permissions"
-  expect_output --from="${_output_otherfile_stat}" "${_expect_otherfile_stat}" "Check if another copied file gets the right permissions"
```

Comment:
There really is no reason for this to be in a helper; it just complicates arg passing.